diff --git a/src/action_tree.rs b/src/action_tree.rs
index 8366c12..7ae66c1 100644
--- a/src/action_tree.rs
+++ b/src/action_tree.rs
@@ -390,10 +390,10 @@ impl ActionTree {
             let mut node = &*self.root.lock() as *const ActionTreeNode;
             for action in &self.history {
                 while (*node).is_chance() {
-                    node = &*(*node).children[0].lock();
+                    node = &*(&(*node).children)[0].lock();
                 }
                 let index = (*node).actions.iter().position(|x| x == action).unwrap();
-                node = &*(*node).children[index].lock();
+                node = &*(&(*node).children)[index].lock();
             }
             &*node
         }
@@ -405,7 +405,7 @@ impl ActionTree {
         unsafe {
             let mut node = self.current_node() as *const ActionTreeNode;
             while (*node).is_chance() {
-                node = &*(*node).children[0].lock();
+                node = &*(&(*node).children)[0].lock();
             }
             &*node
         }
